### PR TITLE
fix: bookings made via API don't include the specified location

### LINKED ISSUE #14341: [CAL-3360] bookings made via API don't include the specified location
The bookings that have come through via API are all booking on cal video but the ones through the site directly actually use zoom as they should.

<sub>From [SyncLinear.com](https://synclinear.com) | [CAL-3360](https://linear.app/calcom/issue/CAL-3360/bookings-made-via-api-dont-include-the-specified-location)</sub>

### LINKED ISSUE #15059: [CAL-3884] Cal Video is used as conferencing app instead of users default app when event's location is not set
Found a bug? Please fill out the sections below. 👍

### Issue Summary

For newly registered users , 3 events are created by default.
User chooses to install zoom as conferencing app , sets zoom as default conferencing app ,
User rejects bulk update option ( without applying bulk update location for events ).
The events show select location in event view section .
When a client books the users event it is expected zoom is used as conferencing app , but cal video is shows as location after booking

### Steps to Reproduce

1. Register a new account on cal.com
2. Install Zoom ( or any other conferencing app )
3. Set newly installed conferencing app as default ( without accepting bulk update )
4. book an event from an alternative account
5. cal video will be used for meeting

Any other relevant information. For example, why do you consider this a bug and what did you expect to happen instead?

### Actual Results

- Cal Video is used

### Expected Results

- The default conferencing app of user should be used ( Zoom in this case ) 

<sub>[CAL-3884](https://linear.app/calcom/issue/CAL-3884/cal-video-is-used-as-conferencing-app-instead-of-users-default-app)</sub>

### PR DESCRIPTION
**Pull request by Shaik-Sirajuddin**
## What does this PR do?

- Fixes #14341 and #15059  (GitHub issue number)
- Fixes CAL-3360 (Linear issue number - should be visible at the bottom of the GitHub issue description)

Before : 
Fix #14341 
Before : [Loom](https://www.loom.com/share/babb9c2bd0a44b25a80ca6fe71045890?sid=c3537ce6-e6a5-4c1a-9dc1-8e61b6c95ad4)
After : [Loom](https://www.loom.com/share/8ca20bfc1d8b483b81489f2e78eac6fc?sid=a870126a-d29d-4e93-a155-3e5a8b724023)

Fix #15059 :
Before : [Loom](https://www.loom.com/share/7973b25c040c40ba83d32a82a9252adb?sid=21ea3820-dc67-4e5a-b958-43ed1ebb8ac9)
After : [Loom](https://www.loom.com/share/f295685fde6e4c23a9cd6e83479e1b74?sid=6493d934-892e-47c2-984b-b45cae3fb166)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected)
- [ ] I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com)
- [ ] I have added or modified automated tests that prove my fix is effective or that my feature works (PRs might be rejected if logical changes are not properly tested)

## How should this be tested?

- #14341
- Create an event in profile with location set to conferencing or option of choice
- Create a booking for just created event via API , leaving location blank or unspecified
- View bookings page , the location used for booking will be the one specified in event or users default
- #15059 
- Create a new user account , a new user account has default events with no location set
- Update default conferencing app in profile settings , reject bulk update event option
- Book an event of newly created account via alternative account
- Booking will use users default conferencing app if events default is not available 
## Checklist

- I haven't checked if my changes generate no new warnings

### CODE REVIEW

**Comment by Shaik-Sirajuddin** [comment]
Loom Videos were commented out earlier.
Video links are now available in description for easier review.

**Review by sean-brydon — DISMISSED** [review]
Hey @Shaik-Sirajuddin I have manually tested this and it works pretty well. 

Do you think you could expand our tests to cover this edge case? 
Off the top of my head `fresh-booking.test.ts` will be the test fill that needs this edge case adding to

**Comment by Shaik-Sirajuddin** [comment]
> Hey @Shaik-Sirajuddin I have manually tested this and it works pretty well.
> 
> Do you think you could expand our tests to cover this edge case? Off the top of my head `fresh-booking.test.ts` will be the test fill that needs this edge case adding to

Hi @sean-brydon
As suggested , I have added tests to cover the cases when location is not specified.
A fix for a failing test case has also been implemented.
PTAL

File: packages/features/bookings/lib/handleNewBooking/test/team-bookings/collective-scheduling.test.ts [hunk 1]
```diff
@@ -1287,7 +1287,7 @@ describe("handleNewBooking", () => {
           expectBookingCreatedWebhookToHaveBeenFired({
             booker,
             organizer,
-            location: OrganizerDefaultConferencingAppType,
+            location: BookingLocations.CalVideo,
```
Comment (Shaik-Sirajuddin) [inline packages/features/bookings/lib/handleNewBooking/test/team-bookings/collective-scheduling.test.ts:1290]: webook responses contains exact location of booking
which is calvideo in this case

**Review by emrysal — APPROVED** [review]
Looks good to me, happy to merge 🙏  - thank you for adding the tests

**Pull request merged**